### ISSUE TITLE
TerminalMenus broken with JuliaVSCode's `execute`

### ISSUE THREAD

**Issue by korsbo** [julia-vscode/julia-vscode#2668]
Hello,

I'm trying to make a nice script-generation wizard using the stdlib `REPL.TerminalMenus` but it does not play nice with the standard code execution commands in julia-vscode. When using `alt+enter` (or ctrl, or shift) you get a terminal menu in the terminal but it does not respond properly to keypresses. Pressing up-arrow for example, will alternately move the terminal menu cursor (correct) and move up among the commands in the REPL history (incorrect). If you instead use `julia: send line or current selection to REPL` then it works just fine. 

![image](https://user-images.githubusercontent.com/16306680/148192515-9c9308cd-e46b-49d1-9567-9277d392bfc9.png)
![image](https://user-images.githubusercontent.com/16306680/148192447-cdfc4da9-8c3f-4ff2-9b7a-837df720426e.png)

MWE:
```julia
using REPL.TerminalMenus
idx = request("Select some things:", MultiSelectMenu(["foo", "bar", "baz"], charset=:unicode))
```

I'm using 
julia-vscode  v1.5.8
julia 1.6

Thanks!

**Comment by pfitzseb** [comment]
Yeah, that's [the classic `readline` bug...](https://github.com/julia-vscode/julia-vscode/issues/785)

It's fairly tricky to convince the REPL to stop reading from stdin during async computations, unfortunately.

**Comment by korsbo** [comment]
Ouch. Ok, thanks for the quick reply! I'll try to find some alternative solution for my little wizard then.
